public/app/plugins/datasource/graphite/datasource.ts

```diff
@@ -178,40 +184,67 @@ export class GraphiteDatasource
   }
 
   query(options: DataQueryRequest<GraphiteQuery>): Observable<DataQueryResponse> {
-    const graphOptions = {
-      from: this.translateTime(options.range.from, false, options.timezone),
-      until: this.translateTime(options.range.to, true, options.timezone),
-      targets: options.targets,
-      format: (options as any).format,
-      cacheTimeout: options.cacheTimeout || this.cacheTimeout,
-      maxDataPoints: options.maxDataPoints,
-    };
+    const streams: Array<Observable<DataQueryResponse>> = [];
 
-    const params = this.buildGraphiteParams(graphOptions, options.scopedVars);
-    if (params.length === 0) {
-      return of({ data: [] });
-    }
+    for (const target of options.targets) {
+      if (target.hide) {
+        continue;
+      } else if (target.fromAnnotations) {
+        streams.push(
+          new Observable((subscriber) => {
+            this.annotationEvents(options.range, target)
+              .then((events) => {
+                return subscriber.next({ data: [toDataFrame(events)] });
+              })
+              .catch((ex) => {
+                return subscriber.error(new Error(ex));
+              })
+              .finally(() => subscriber.complete());
+          })
+        );
+      } else {
+        // handle the queries
+        const graphOptions = {
+          from: this.translateTime(options.range.from, false, options.timezone),
+          until: this.translateTime(options.range.to, true, options.timezone),
+          targets: options.targets,
+          format: (options as any).format,
```

Comment:
So I see that there is no `format` field in options. But where is this coming from?

Reply:
If that is necessary I think we should find a way to make types happy about it :)

Reply:
Ahh, yes, this is old code that I had to put back into this PR due to an incorrect merge on my part. Let me look into the history of this 'format' though...

Ok, it was added in 2019 here
https://github.com/grafana/grafana/commit/026d13469fa8a9ed3ae24958f20c864b7e9a4ede#diff-c27e82d1dbdf51beeab4bc60a1546fce8a7ffec7a7905f78f4fee06375ab05cb

Reply:
It is nowhere in the DataQueryRequest I can see. I vote on removing it, but just to check, @ryantxu and @torkelo, do you have any thoughts on removing ```format: (options as any).format``` in graphiteOptions? I can't find anywhere where the format is defined. Thank you!

Reply:
Found it! 'format' is set here, I added a type GraphiteRequestOptions and made types happy.

https://github.com/grafana/grafana/blob/92d394b31069f57fbf50078bf9c72ab31bbe708e/public/app/plugins/datasource/graphite/datasource.ts#L864